js/ui/form/ui.form.layout_manager.js

```diff
@@ -744,6 +747,20 @@ var LayoutManager = Widget.inherit({
         return validationRules;
     },
 
+    _addWrapperInvalidClass: function(editorInstance) {
+        var wrapperClass = "." + FIELD_ITEM_CONTENT_WRAPPER_CLASS;
+        if(/material/.test(themes.current())) {
+            var toggleInvalidClass = function(e) {
+                $(e.element).parents(wrapperClass)
+                    .toggleClass(INVALID_CLASS, e.event.type === "focusin" && !e.component.option("isValid"));
```

Comment:
Use `typeUtils.isDefined`. Someone can create a non-editor widget and you will assign the `INVALID_CLASS` to them because `isValid` is `undefined`